common/src/main/java/com/microsoft/identity/common/internal/controllers/BrokerMsalController.java

```diff
@@ -1170,6 +1170,14 @@ private void verifyTokenParametersAreSupported(@NonNull final TokenCommandParame
                     "The min broker protocol version for PopAuthenticationSchemeWithClientKey should be equal or more than 11.0."
                             + " Current required version is set to: " + parameters.getRequiredBrokerProtocolVersion());
         }
+
+        // Nested app auth is only supported when required protocol version is >=15
+        if ((parameters.getChildClientId() != null || parameters.getChildRedirectUri() != null)
```

Comment:
>(parameters.getChildClientId() != null || parameters.getChildRedirectUri() != null) [](http://example.com/codeflow?start=12&length=83)

Should we fail the request if only one of them is set (invalid state)?

Reply:
yes, I think it makes sense to fail the request immediately here if only 1 of them is set.